src/swarm-scenario/Swarm/Game/Scenario/Style.hs

```diff
@@ -38,17 +40,36 @@ instance FromJSON StyleFlag where
 instance ToJSON StyleFlag where
   toJSON = genericToJSON styleFlagJsonOptions
 
--- | Hexadecimal color notation.
--- May include a leading hash symbol (see 'Data.Colour.SRGB.sRGB24read').
-newtype HexColor = HexColor Text
-  deriving (Eq, Ord, Show, Generic, FromJSON, ToJSON)
+-- | A color, parsed from hexadecimal notation.  May include a leading
+--   hash symbol (see 'Data.Colour.SRGB.sRGB24read').
+newtype HexColor = HexColor {getHexColor :: Kolor}
+  deriving (Eq, Show, Generic)
+
+instance Ord HexColor where
+  -- There is no Ord instance for Colour a, but we need one to use
+  -- with OccurrenceEncoder, so we make our own.
+  --
+  -- We use toRGB here since it does no conversions whatsoever, it
+  -- simply unpacks the raw color data into an RGB triple.  For the
+  -- purposes of an Ord instance, it doesn't matter: we just want a
+  -- consistent way to put a total ordering on colors as fast as
+  -- possible.
+  compare (HexColor (toRGB -> RGB r1 g1 b1)) (HexColor (toRGB -> RGB r2 g2 b2)) =
+    compare r1 r2 <> compare g1 g2 <> compare b1 b2
```

Comment:
would it be simpler to use [3-tuples](https://hackage.haskell.org/package/ghc-prim-0.13.0/docs/GHC-Tuple.html#t:Ord)?
```
compare (r1, g1, b1) (r2, g2, b2)
```